### ISSUE TITLE
Slower cursor updates in 3.6.x

### ISSUE THREAD

**Issue by baskerville** [facebook/immutable-js#284]
I'm experiencing a performance degradation with 3.6.x.

It seems to be caused by the following function:
```
function newKeyPath(head, tail) {
	return Seq(head).concat(tail).toArray();
}
```

If I replace it with:
```
function newKeyPath(head, tail) {
	return head.concat(tail);
}
```
performances are back to normal.

**Comment by leebyron** [comment]
Can you explain the performance issue further or share a benchmark?

The replacement will not work correctly if your key-paths are Iterables instead of Array, and I would like to preserve that behavior.

If you share a benchmark I can attempt to solve for the performance issue while maintaining the iterable keypath property.

**Comment by baskerville** [comment]
```
var Immutable = require("immutable");

var a = [1, 2, 3, 5, 6, 7, 8, 9, 10],
    fns = {
    	js: function (a) {
    		return a.concat(a);
		},
		im: function (a) {
			return Immutable.Seq(a).concat(a).toArray();
		}
	},
	scores = [];

for (var name in fns) {
	var fn = fns[name],
	    before = Date.now();
	for (var i = 0; i < 3000; i++) {
		fn(a);
	}
	var delta = Date.now() - before;
	scores.push(delta);
}

console.log(scores[1]/scores[0]);
```

**Comment by leebyron** [comment]
Thanks

**Comment by leebyron** [comment]
Released as v3.6.2. Thanks for the quick report.
